modules/apt.sh

```diff
@@ -43,3 +51,38 @@ install_package_if_missing_file() {
         check_result apt_get install "$package"
     fi
 }
+
+_apt_write_list_file() {
+    local repo_file="$1"
+    local repo_url="$2"
+
+    cat >"$repo_file" <<EOF
+deb https://${repo_url}/ubuntu ${SERIES} main
+# deb-src https://${repo_url}/ubuntu ${SERIES} main
+EOF
+}
+
+_apt_add_auth() {
+    local repo_url="$1"
+    local credentials="$2"
+
+    local login password
+    login=$(echo "$credentials" | cut -d: -f1)
+    password=$(echo "$credentials" | cut -d: -f2)
+    [ -f "$APT_AUTH_FILE" ] || touch "$APT_AUTH_FILE"
+    chmod 600 "$APT_AUTH_FILE"
+    echo "machine ${repo_url}/ubuntu/ login ${login} password ${password}" \
+         >>"$APT_AUTH_FILE"
```

Comment:
Once we check if esm is already enabled before calling this, we should be safe(r) and be able to avoid multiple identical lines in auth.conf, but maybe we should add a second check and only add our line if it doesn't exist already. Maybe just check for the url (careful with substring matches).

Reply:
I don't think we should go too deep with validating the content of the file. The user could always put crap in the file, we should just expect what we put there. As you said, this will be safer once we check before adding. Note that disable would remove all lines for the host, if there are many